Bundle/DependencyInjection/EnqueueAdapterExtension.php

```diff
@@ -25,5 +25,9 @@ public function load(array $configs, ContainerBuilder $container)
     {
         $loader = new YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.yml');
+
+        if (!$container->hasDefinition('messenger.transport.serializer')) {
+            throw new \LogicException('The default Messenger serializer cannot be enabled as the Serializer support is not available. Try enabling it or running "composer require symfony/serializer-pack".');
+        }
```

Comment:
I'm not sure if this will work - because the `messenger.transport.serializer` comes from FrameworkBundle and the `$container` passed to each Extension class is "empty" - sorta of "sandboxed". In other words, this `$container` will never have this service. I think this needs to be done in a compiler pass.

Reply:
I will check and if not will implement the compiler pass no problem :+1:

Reply:
@weaverryan you where right about the sandboxed ContainerBuilder so I implemented your original suggestion.